zephyr/lib/cpu.c

```diff
@@ -99,6 +125,29 @@ void cpu_notify_state_exit(enum pm_state state)
 #endif
 
 #ifdef CONFIG_ADSP_IMR_CONTEXT_SAVE
+		struct ipc_comp_dev *icd;
+		struct list_item *clist;
+		struct processing_module *mod;
+		struct copier_data *cd;
+		struct dai_data *dd;
+
+		/* Resume all DAI components after exiting D3 state. */
+		list_for_item(clist, &ipc_get()->comp_list) {
+			icd = container_of(clist, struct ipc_comp_dev, list);
+			if (icd->type != COMP_TYPE_COMPONENT)
+				continue;
+
+			if (dev_comp_type(icd->cd) == SOF_COMP_DAI) {
+				mod = comp_mod(icd->cd);
+				cd = module_get_private_data(mod);
+				dd = cd->dd[0];
+				if (dai_probe(dd->dai->dev) < 0) {
+					tr_err(&zephyr_tr, "DAI resume failed, type %d index %d",
+					       dd->dai->type, dd->dai->index);
+				}
+			}
+		}
+
```

Comment:
ack with @dbaluta  - I think this makes sense to have an API for DAI suspend/resume as these are handy utilities. We could have these in dai-zephyr.c.

Reply:
e.g. ```dai_suspend_all()```

Reply:
Implemented static helper functions `suspend_dais()` and `resume_dais()` in cpu.c for better code readability. I've opted not to extend the DAI interface to keep changes minimal and localized, as the current DAI management aligns with SOF's power management needs without additional API modifications.